File: backend/core/src/modules/relationships/relationships.service.ts
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { ProjectAccessService } from '../../common/access/project-access.service';
import { ProjectScopedCrudService } from '../../common/access/project-scoped-crud.service';
import { Relationship } from '../../entities';

@Injectable()
export class RelationshipsService extends ProjectScopedCrudService<Relationship> {
  constructor(
    @InjectRepository(Relationship)
    repo: Repository<Relationship>,
    projectAccessService: ProjectAccessService,
  ) {
    super(repo, projectAccessService);
  }
}

File: backend/core/src/main/java/store/pjcloud/cowrite/core/service/RelationshipsService.java
package store.pjcloud.cowrite.core.service;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.stereotype.Service;
import store.pjcloud.cowrite.core.common.ProjectScopedCrudService;
import store.pjcloud.cowrite.core.entity.Relationship;
import store.pjcloud.cowrite.core.repository.RelationshipRepository;

@Service
public class RelationshipsService extends ProjectScopedCrudService<Relationship> {
    public RelationshipsService(RelationshipRepository repository,
                                ProjectAccessService projectAccessService,
                                ObjectMapper objectMapper) {
        super(repository, projectAccessService, objectMapper, Relationship.class);
    }
}
